idb/postgres/postgres.go

```diff
@@ -658,9 +658,12 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 	if tf.RekeyTo != nil && (*tf.RekeyTo) {
 		whereParts = append(whereParts, "(t.txn -> 'txn' -> 'rekey') IS NOT NULL")
 	}
+	if tf.ReturnRootTxnsOnly {
+		whereParts = append(whereParts, "t.txid IS NOT NULL")
+	}
 
 	// If returnInnerTxnOnly flag is false, then return the root transaction
-	if !tf.ReturnInnerTxnOnly {
+	if !(tf.ReturnInnerTxnOnly || tf.ReturnRootTxnsOnly) {
```

Comment:
The `ReturnInnerTxnOnly` variable name does not really represent what this is doing--it makes it much harder to reason about these conditions.

Let's update the name on the develop branch--avoiding a bunch of refactoring on this hotfix branch seems more important.

I've updated the most recent patch to represent what we want, which is that if neither of these options is enabled we will fetch `root.txn` and _also_ if neither is enabled we'll add the JOIN which references the intra index.